src/Text/Layout/Table/Cell.hs

```diff
@@ -37,17 +37,22 @@ class Cell a where
     -- the predicate matches.
     measureAlignment :: (Char -> Bool) -> a -> AlignInfo
 
+    -- | Create an empty cell.
+    -- This should satisfy `buildCell emptyCell = mempty`.
+    emptyCell :: a
```

Comment:
I'm not sure if I asked this before but what is the point of it?  If this is just there to be used once in an internal algorithm then I have to question its existence.

Reply:
It's needed for padding, the same as the Monoid constraint on `colsG` and friends. Whenever we have more than one `Row` in a `RowGroup`, we display the row header only for the first row, and need to pad out the remaining rows with something. That is the role that `emptyCell` fills.

Without something to put there we would not be able to create row headers, at least not without doing something else, like repeating the row headers for every row in the group, which is not desired.

Reply:
Another thing to note: `Cell` is pretty much defined as being a typeclass which maps to various `Monoid`, though `buildCell` (since `StringBuilder` is a subclass of `Monoid`). In this context, `emptyCell` has the fairly natural and lawful definition of being an element which maps to `mempty` under `buildCell`.

Reply:
Thanks!